jbinding-java/src/net/sf/sevenzipjbinding/SevenZip.java

```diff
@@ -643,9 +679,15 @@ private static void applyTemporaryArtifacts(File sevenZipJBindingTmpDir, List<Fi
     private static void loadNativeLibraries(List<File> libraryList) throws SevenZipNativeInitializationException {
         // Load native libraries in to reverse order
         for (int i = libraryList.size() - 1; i != -1; i--) {
-            String libraryFileName = libraryList.get(i).getAbsolutePath();
+            String libraryFileName = null;
             try {
-                System.load(libraryFileName);
+                if (System.getProperty("java.vendor", "unknown").equals("The Android Project")) {
+                    libraryFileName = libraryList.get(i).getName();
+                    System.loadLibrary(libraryFileName);

```

Comment:
`System.loadLibrary()` is all that's needed on Android. This class should be simplified to allow Android to use this method directly to load the native library.